src/main/java/edu/harvard/iq/dataverse/DatasetServiceBean.java

```diff
@@ -165,20 +165,22 @@ private List<Dataset> findByOwnerId(Long ownerId, boolean onlyPublished) {
     }
 
     public List<Long> findIdsByOwnerId(Long ownerId) {
-        return findIdsByOwnerId(ownerId, false);
+        return findIdsByOwnerId(ownerId, false, false);
     }
 
-    private List<Long> findIdsByOwnerId(Long ownerId, boolean onlyPublished) {
+    public List<Long> findIdsByOwnerId(Long ownerId, boolean onlyPublished, boolean includeHarvested) {
         List<Long> retList = new ArrayList<>();
-        if (!onlyPublished) {
+        if (!onlyPublished && includeHarvested) {
             return em.createNamedQuery("Dataset.findIdByOwnerId")
                     .setParameter("ownerId", ownerId)
                     .getResultList();
         } else {
             List<Dataset> results = em.createNamedQuery("Dataset.findByOwnerId")
                     .setParameter("ownerId", ownerId).getResultList();
             for (Dataset ds : results) {
-                if (ds.isReleased() && !ds.isDeaccessioned()) {
+                if (onlyPublished && ds.isReleased() && !ds.isDeaccessioned()) {
+                    retList.add(ds.getId());
+                } else if (includeHarvested && ds.isHarvested()) {
```

Comment:
If onlyPublished=true and includeHarvested=true, this doesn't include harvested because of the else?

Reply:
OK - makes sense - could you add a comment about that?

Reply:
I've refactored the code a bit and added some comments, I think it makes more sense now: https://github.com/IQSS/dataverse/pull/11555/commits/7eee55eefe04b5191c7cfa0cf652791608ef56c8